src/main/java/com/nha/abdm/wrapper/common/RequestManager.java

```diff
@@ -11,16 +11,24 @@
 @Component
 public class RequestManager {
 
-  @Autowired SessionManager sessionManager;
+  private WebClient webClient;
+
+  public RequestManager(
+      @Value("${gatewayBaseUrl}") final String gatewayBaseUrl, SessionManager sessionManager) {
+    webClient =
+        WebClient.builder()
+            .baseUrl(gatewayBaseUrl)
+            .defaultHeaders(
+                httpHeaders -> httpHeaders.addAll(sessionManager.setGatewayRequestHeaders()))
+            .build();
+  }
 
   public <T> ResponseEntity<ObjectNode> fetchResponseFromPostRequest(String uri, T request) {
     WebClient.Builder webClientBuilder = WebClient.builder();
```

Comment:
This line `WebClient.Builder webClientBuilder = WebClient.builder();` can be removed now.

Reply:
yes, realized. Removed the line.